### ISSUE TITLE
 BEAST_AFFINITY merit, pet shall exceed master lvl.

### ISSUE THREAD

**Issue by benfolka** [DarkstarProject/darkstar#6266]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 30191004_0

**_Source Branch_** (master/stable) **:** master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** Since pet not reciving any exp fix has been implemented, This wont cause any problems. This behaviour is retail and shall be same in DSP servers. Pets shall exceed master level with beast affinity merits.

**Comment by zynjec** [comment]
I'm not sure I follow the issue, BEAST_AFFINITY already allows the pet to go above the masters level unless you've experienced it not working?

https://github.com/DarkstarProject/darkstar/blob/43698546fcb01ef1eae1adb2ceac1f57c5f043b0/src/map/utils/petutils.cpp#L1652

**Comment by gori123** [comment]
// And cap it to the master's level.
            if (highestLvl > PMaster->GetMLevel())
            {
                highestLvl = PMaster->GetMLevel();
            }

**Comment by gori123** [comment]
merit works but then there are lines that drops pet level again to masters level,
which is not retail default

**Comment by zynjec** [comment]
Fair enough, just grepped for BEAST_AFFINITY and didn't see the code under it. Seems easy enough to fix.

**Comment by TeoTwawki** [comment]
there is supposed to be some capping, but its need to be done before merit and other mods apply, not after.

got stuffed in wrong place.
